lib/MySQL_Thread.cpp

```diff
@@ -1137,52 +1399,32 @@ void MySQL_Thread::process_data_on_data_stream(MySQL_Data_Stream *myds, unsigned
 						myds->sess->set_unhealthy();
 					}
 				}
-
-/*
-	      if (myds->active==FALSE) {
-					mypolls.remove_index_fast(n);
-					proxy_debug(PROXY_DEBUG_NET,1, "Session=%p, DataStream=%p -- Deleting FD %d\n", myds->sess, myds, myds->fd);
-					//myds->shut_hard();
-					MySQL_Session *sess=myds->sess;
-					if (
-						(sess->server_myds==myds)
-						&&
-						(myds->myds_type==MYDS_BACKEND)
-						&&
-						(myds->DSS==STATE_READY)
-					) {
-						if (sess->mybe->myconn) {
-							MyHGM->destroy_MyConn_from_pool(sess->mybe->myconn);
-							sess->mybe->myconn=NULL;
-						}
-						// This is a failed backend, let's try to save the session
-						return;
-					}
-
-					sess->healthy=0;
-					if (sess->client_myds==myds) {
-						sess->client_myds=NULL;
-						delete myds;
-					}
-					if (sess->server_myds==myds) {
-						sess->server_myds=NULL;
-					}
-					//delete myds;
-					//myds=NULL; // useless?
-// FIXME
-//  	   		if (sess->client_myds==NULL && sess->server_myds==NULL) {
-//						mysql_sessions->remove_fast(sess);
-//						delete sess;
-//						continue;
-//					}
-				}
-*/
 }
 
 
 void MySQL_Thread::process_all_sessions() {
 	unsigned int n;
 	int rc;
+	bool sess_sort=mysql_thread___sessions_sort;
```

Comment:
eheheh , right!
Actually, sessions_sort is a global variable, and should be documented in the documentation.